pkg/proxy/helpers.go

```diff
@@ -0,0 +1,67 @@
+package proxy
+
+import (
+	"context"
+
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/types"
+	cli "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// ClusterProxyName is the name of the global config proxy.
+	ClusterProxyName = "cluster"
+)
+
+var (
+	// ClusterProxyKey is the key for the cluster proxy.
+	clusterProxyKey = types.NamespacedName{Name: ClusterProxyName}
+)
+
+// CheckProxyResource returns true if the deployment with the given namespacedname
+// needs to have its environment variables update due to a change in the cluster proxy.
+func CheckProxyResource(client cli.Client, name, namespace string) (bool, error) {
+	// Check if the Proxy API exists.
+	if !IsAPIAvailable() {
+		return false, nil
+	}
+
+	// Get the Deployment with the given namespacedname.
+	deployment := &apps.Deployment{}
+	key := types.NamespacedName{Namespace: namespace, Name: name}
+	if err := client.Get(context.TODO(), key, deployment); err != nil {
+		return false, err
+	}
+
+	// Check if the deployment environment variables are in sync with the proxy.
+	if NeedsUpdate(*deployment) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+// NeedsUpdate return true if the deployment is out of sync with the proxy environment variables.
+func NeedsUpdate(registryDeployment apps.Deployment) bool {
+	// Get the list of environment variables defined in the registry deployment
+	actualEnvVars := registryDeployment.Spec.Template.Spec.Containers[0].Env
+	if len(actualEnvVars) == 0 {
+		return true
+	}
+
+	desiredEnvVars := GetOperatorEnvVars()
+
+	if len(desiredEnvVars) != len(actualEnvVars) {
+		return true
+	}
+
+	// Marketplace takes ownership of env vars, and expects them
+	// to be in the order returned by GetOperatorEnvVars.
```

Comment:
This is risky. There is no guarantee even then if the ordering will be the same. It is safer to sort and then compare. Take a look at [cache.IsEntryStale()](https://github.com/operator-framework/operator-marketplace/blob/master/pkg/catalogsourceconfig/cache.go#L77).

Reply:
Understood, I will make sure to sort and then compare the two lists.